src/resolver.rs

```diff
@@ -801,6 +801,29 @@ impl<'i> TypeAnnotator<'i> {
                     StatementAnnotation::value(expected_type.get_name()),
                 );
                 self.update_expected_types(expected_type, initializer);
+
+                // handle annotation for array of struct
+                if let DataTypeInformation::Array {
+                    inner_type_name, ..
+                } = expected_type.get_type_information()
+                {
+                    let struct_type = self
+                        .index
+                        .get_effective_type_or_void_by_name(inner_type_name);
+                    if let DataTypeInformation::Struct { .. } = struct_type.get_type_information() {
```

Comment:
here maybe also `struct_type.get_type_information().is_struct()` is a bit more expressive